### PR TITLE
22.2 task attachments kovalets

### PR DESCRIPTION
**Pull request by AndriyKovalets**

### CODE REVIEW

File: src/app/core/models/task/taskAttachment.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export class TaskAttachment {
+    id: number;
+    name: string;
+}
```
Comment (FeloniukM) [inline src/app/core/models/task/taskAttachment.ts]: Please add a space at the end of the file

File: src/app/core/models/task/uploadTaskAttachments.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export class UnloadTaskAttachment {
+    workspaceId: number;
+    taskId: number;
+    attachment: File;
+}
```
Comment (FeloniukM) [inline src/app/core/models/task/uploadTaskAttachments.ts]: There is also a gap

File: src/app/core/services/task.service.ts [hunk 3]
```diff
@@ -48,6 +50,36 @@ export class TaskService {
...
+    const options = {
+      headers: this.httpOption.headers,
+      responseType: 'Blob' as 'json'
+    }
+
+    return this.http.get<File>(this.taskServiceUrl + 'task/workspace/' + workspaceId + '/attachment/'+ attachmentId, options);
+  }
+
+  uploadAttachment(attachment:  UnloadTaskAttachment): Observable<TaskAttachment> {
+
```
Comment (FeloniukM) [inline src/app/core/services/task.service.ts:69]: Is a space needed here?

File: src/app/presentation/components/workspace-components/task-components/workspace-task-attachments/workspace-task-attachments.component.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  getAttachmentList() {
+    this.taskService.getAttachmentList(this.workspaceId, this.taskId).subscribe((data) =>{
+      this.attachments = data;
+    });
+  }
+
+  uploadAttachment(event: any) {
+
```
Comment (FeloniukM) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-attachments/workspace-task-attachments.component.ts:38]: Is a space needed here?

File: src/app/presentation/components/workspace-components/task-components/workspace-task-attachments/workspace-task-attachments.component.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  downloadFile(attachmentId: number) {
+    this.taskService.getAttachment(this.workspaceId, attachmentId).subscribe(file => {
+      saveAs(file, file.name);
+    });
+  }
+
+  getAttacmentName(name: string): string {
+
```
Comment (FeloniukM) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-attachments/workspace-task-attachments.component.ts:68]: Is a space needed here?

**Review by FeloniukM — APPROVED** [review]

**Review by herakros — APPROVED** [review]

**Pull request merged**